### ISSUE TITLE
 Finding Gene Sense Strand - 21U-RNAs

### ISSUE THREAD

**Issue by Gschindelman** [WormBase/website#4960]
the following came in through help desk

Hello,

I was performing a search using the QueryBuilder under WormMine to create a list of all the locations of 21U-RNAs (of which there are about 15000) in C. elegans.  I used the constraint filter "CONTAINS" "21ur" under "Gene Name", and asked the program to show "Location"--> "Strand"; "Start"; and "End", as well as "Located On Wormbase ID" to find the chromosome each sequence was located on.  Unfortunately, for all the 21U-RNAs, the Location-->Strand does not yield whether each sequence is on the +1 or -1 strand, but always displays a "0" for that information, giving me no information for what strand the sense information is encoded.  I imagine that there is a way for me to determine the strand using QueryBuilder, but I cannot figure out how.  

Your help is very greatly appreciated,
Darius Mostaghimi

Department of Biology
Johns Hopkins University

**Comment by Gschindelman** [comment]
dmostag1@jhu.edu
to follow the answer to your question please follow
https://github.com/WormBase/website/issues/4960

**Comment by Paul-Davis** [comment]
Dear Darius,

Thank you for notifying us of the strand issue in or data mining 
platform. While the developers look into the issue I can point you to an 
alternative source of this data.

If you download our gff3 data file:

ftp://ftp.wormbase.org/pub/wormbase/releases/WS253/species/c_elegans/PRJNA13758/c_elegans.PRJNA13758.WS253.annotations.gff3.gz

All of the gene co-ordinate data is available with the strand.

Example unix command:
zcat elegans.processed.gff3.gz | grep 21ur | grep -v 
absolute_pmap_position | awk '{if ($3 == "gene") print $0;}'

Example data:

I    WormBase    gene    115656    115676    .    -    .    ID=Gene:WBGene00171818;Name=WBGene00171818;locus=21ur-13877;sequence_name=F53G12.14;biotype=piRNA;Alias=21ur-13877,F53G12.14
I    WormBase    gene    272603    272623    .    -    .    ID=Gene:WBGene00169222;Name=WBGene00169222;locus=21ur-15400;sequence_name=C53D5.8;biotype=piRNA;Alias=21ur-15400,C53D5.8
I    WormBase    gene    272607    272627    .    -    .    ID=Gene:WBGene00166448;Name=WBGene00166448;locus=21ur-13405;sequence_name=C53D5.7;biotype=piRNA;Alias=21ur-13405,C53D5.7
I    WormBase    gene    463304    463324    .    +    .    ID=Gene:WBGene00166632;Name=WBGene00166632;locus=21ur-8411;sequence_name=W04C9.7;biotype=piRNA;Alias=21ur-8411,W04C9.7
I    WormBase    gene    1093137    1093157    .    -    .    ID=Gene:WBGene00174760;Name=WBGene00174760;locus=21ur-15525;sequence_name=R06A10.6;biotype=piRNA;Alias=21ur-15525,R06A10.6
I    WormBase    gene    1487607    1487627    .    -    .    ID=Gene:WBGene00170789;Name=WBGene00170789;locus=21ur-13232;sequence_name=F47G6.5;biotype=piRNA;Alias=21ur-13232,F47G6.5


I hope this helps

Paul



On 15/07/2016 17:54, Darius Mostaghimi wrote:
>
> Hello,
>
>
> I was performing a search using the QueryBuilder under WormMine to 
> create a list of all the locations of 21U-RNAs (of which there are 
> about 15000) in C. elegans.  I used the constraint filter "CONTAINS" 
> "21ur" under "Gene Name", and asked the program to show "Location"--> 
> "Strand"; "Start"; and "End", as well as "Located On Wormbase ID" to 
> find the chromosome each sequence was located on.  Unfortunately, for 
> all the 21U-RNAs, the Location-->Strand does not yield whether each 
> sequence is on the +1 or -1 strand, but always displays a "0" for that 
> information, giving me no information for what strand the 
> sense information is encoded.  I imagine that there is a way for me to 
> determine the strand using QueryBuilder, but I cannot figure out how.
>
>
> Your help is very greatly appreciated,
>
> Darius Mostaghimi
>
>
> Department of Biology
>
> Johns Hopkins University
>
> -- 
> -- 
> You received this message because you are subscribed to the "WormBase: 
> help" group.
> To post to this group, send email to help@wormbase.org
> To unsubscribe from this group, send email to
> help+unsubscribe@wormbase.org
> For more options, visit this group at
> http://groups.google.com/a/wormbase.org/group/help?hl=en
>


-- 
===========================================
WormBase
European Bioinformatics Institute (EMBL-EBI)
European Molecular Biology Laboratory
Wellcome Trust Genome Campus
Hinxton
Cambridge CB10 1SD
United Kingdom
===========================================

**Comment by chris-grove** [comment]
There is apparently a problem with the 21u RNAs (a.k.a. piRNAs) chromsome location in WormMine. I can successfully pull up the list of all piRNAs with the following query:

```<query name="" model="genomic" view="Gene.primaryIdentifier Gene.secondaryIdentifier Gene.symbol Gene.transcripts.method Gene.transcripts.symbol" longDescription="" sortOrder="Gene.primaryIdentifier asc">
  <constraint path="Gene.transcripts.method" op="=" value="piRNA"/>
</query>```

but as soon as I try to add chromosome or chromosome location information I get an error. For example, this query:

```<query name="" model="genomic" view="Gene.primaryIdentifier Gene.secondaryIdentifier Gene.symbol Gene.transcripts.method Gene.transcripts.symbol Gene.chromosome.primaryIdentifier" longDescription="" sortOrder="Gene.primaryIdentifier asc">
  <constraint path="Gene.transcripts.method" op="=" value="piRNA"/>
</query>```

results in the following error message:

![screen shot 2016-07-18 at 12 48 03 pm](https://cloud.githubusercontent.com/assets/611382/16928134/e585e0e4-4ce5-11e6-88d9-46ceca08e121.png)

**Comment by Paul-Davis** [comment]
Yup I can confirm that you can get Transcript data out of the mine, but any filter that constrains to 21ur/piRNA etc. gives 0 results. Looks to be an issue in the data loader?
